lib/Conversion/StableHLOToTTIR/StableHLOToTTIRPatterns.cpp

```diff
@@ -750,17 +750,70 @@ class StableHLOToTTIRBroadcastInDimOpConversionPattern
       return legalityResult;
     }
 
+    auto inputType = mlir::cast<RankedTensorType>(
+        getTypeConverter()->convertType(srcOp.getOperand().getType()));
+
     auto outputType = mlir::cast<RankedTensorType>(
         getTypeConverter()->convertType(srcOp.getResult().getType()));
-    tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
-        srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
 
-    mlir::ArrayAttr dimArg =
-        rewriter.getI64ArrayAttr(adaptor.getBroadcastDimensions());
+    if (inputType.getRank() == outputType.getRank()) {
+      // unsqueeze is not needed, proceed to converting to broadcast
 
-    rewriter.replaceOpWithNewOp<mlir::tt::ttir::BroadcastOp>(
-        srcOp, getTypeConverter()->convertType(outputTensor.getType()),
-        Value(adaptor.getOperand()), Value(outputTensor), dimArg);
+      tensor::EmptyOp outputTensor = rewriter.create<tensor::EmptyOp>(
+          srcOp.getLoc(), outputType.getShape(), outputType.getElementType());
+
+      mlir::ArrayAttr dimArg =
+          rewriter.getI64ArrayAttr(adaptor.getBroadcastDimensions());
+
+      rewriter.replaceOpWithNewOp<mlir::tt::ttir::BroadcastOp>(
+          srcOp, getTypeConverter()->convertType(outputTensor.getType()),
+          Value(adaptor.getOperand()), Value(outputTensor), dimArg);
+    } else {
+
+      std::vector<int64_t> UnsqueezeShape;
+      for (unsigned int i = 0; i < inputType.getRank(); i++) {
+        UnsqueezeShape.push_back(inputType.getDimSize(i));
+      }
+
+      for (unsigned int i = inputType.getRank(); i < outputType.getRank();
```

Comment:
This doesn't look right to me. For broadcast_in_dims you need to insert singleton dimensions where needed to align with broadcast dims. So for example if `input_tensor` is `2x4` and `broadcast_dims` is `1, 2` and `output_tensor` is `2x2x4x2` you first need to reshape to `1x2x4x1`. This will `unsqueeze` to `1x1x2x4`.

Reply:
Yes that's right, I have updated the PR to catch this case, please review.